### DISCUSSION TITLE
How to set training_data/systems and validation_data/systems if I set set_size=some exact number when I prepare the data?

### DISCUSSION THREAD

**Discussion by Lihuigao21** [deepmodeling/deepmd-kit#2512] [Q&A]
For example, I run MD for 5000 steps and I want to 1000 steps of them to be  validation_data, so I set set_size=1000. Then I get set.000 to set.004. Then how can I set training_data/systems and validation_data/systems in input.json? If I delete all the "validation_data" part , I find that the lcurve.out will miss some rows. Thanks for helping me.

I know that if I set set_size=1000 then it is default to regard set.004 as validation_data. But how to let deepmd use this "default" without do harm to lcurve.out?

My input.json:
`{
    "_comment": " model parameters",
    "model": {
	"type_map":	["Ti", "O", "H.66", "H1.33", "H"],
	"descriptor" :{
	    "type":		"se_e2_a",
	    "sel":		[40,70,10,15,15],
	    "rcut_smth":	0.50,
	    "rcut":		6.00,
	    "neuron":		[25, 50, 100],
	    "resnet_dt":	false,
	    "axis_neuron":	16,
	    "seed":		983,
	    "_comment":		" that's all"
	},
	"fitting_net" : {
	    "neuron":		[240, 240, 240],
	    "resnet_dt":	true,
	    "seed":		1,
	    "_comment":		" that's all"
	},
	"_comment":	" that's all"
    },

    "learning_rate" :{
	"type":		"exp",
	"decay_steps":	5000,
	"start_lr":	0.001,	
	"stop_lr":	3.51e-8,
	"_comment":	"that's all"
    },

    "loss" :{
	"type":		"ener",
	"start_pref_e":	0.02,
	"limit_pref_e":	1,
	"start_pref_f":	1000,
	"limit_pref_f":	1,
	"start_pref_v":	0,
	"limit_pref_v":	0,
	"_comment":	" that's all"
    },

    "training" : {
	"training_data": {
	    "systems":		["t=300/", "t=100/", "t=500/", "t=100ts/", "t=300ts/", "t=500ts/", "t=100dis/", "t=300dis/", "t=500dis/"],
	    "batch_size":	"auto",
	    "_comment":		"that's all"
	},

	"numb_steps":	1000000,
	"seed":		10,
	"disp_file":	"lcurve.out",
	"disp_freq":	100,
	"save_freq":	1000,
	"_comment":	"that's all"
    },    

    "_comment":		"that's all"
}
`

**Comment by wanghan-iapcm** [comment]
split the training and validation sets in different folders and refer them by `training_data` and `validation_data`.

**Reply by Lihuigao21** [reply]
Great idea, I will have a try.
